pkg/operation/common/utils.go

```diff
@@ -419,6 +420,87 @@ func DeleteReserveExcessCapacity(ctx context.Context, k8sClient client.Client) e
 	return client.IgnoreNotFound(k8sClient.Delete(ctx, priorityClass))
 }
 
+// DeleteOldLoggingStack deletes all resource of the old EFK logging stack in the given namespace.
+// TODO: Should be remove on the next garden release
+func DeleteOldLoggingStack(ctx context.Context, k8sClient client.Client, namespace string) error {
+	if k8sClient == nil {
+		return errors.New("must provide non-nil kubernetes client to common.DeleteLoggingStack")
+	}
+
+	// Delete the resources below that match "garden.sapcloud.io/role=logging"
+	lists := []runtime.Object{
+		&corev1.ConfigMapList{},
+		&batchv1beta1.CronJobList{},
+		&rbacv1.ClusterRoleList{},
+		&rbacv1.ClusterRoleBindingList{},
+		&rbacv1.RoleList{},
+		&rbacv1.RoleBindingList{},
+		&appsv1.DeploymentList{},
+		&autoscalingv2beta1.HorizontalPodAutoscalerList{},
+		&extensionsv1beta1.IngressList{},
+		&corev1.SecretList{},
+		&corev1.ServiceAccountList{},
+		&corev1.ServiceList{},
+		&appsv1.StatefulSetList{},
+		&corev1.PersistentVolumeClaimList{},
+	}
+
+	labels := []map[string]string{
+		map[string]string{
+			"app": "fluentd-es",
+		},
+		map[string]string{
+			"app": "elasticsearch-logging",
+		},
+		map[string]string{
+			"app": "curator",
+		},
+		map[string]string{
+			"app": "kibana-logging",
+		},
+	}
+
+	for _, list := range lists {
+		for _, value := range labels {
+			if err := deleteList(ctx, k8sClient, list, namespace, "app", value["app"]); err != nil {
+				return err
+			}
+		}
+	}
```

Comment:
Can't you simply enumerate the objects that shall be deleted in a list and then iterate over it?
This way we would only execute a few `Delete()` calls and save A LOT of `List` calls that you are performing here.
Ref https://github.com/gardener/gardener/pull/2478/files#diff-fdd658646d41dc03a2e041468d0c12ccR349-R396 
You are already doing it for the network policies :)